File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        vector<int> suff(arr.size());
        suff[arr.size()-1] = -1;
        for (int i = arr.size()-1; i>=1; i--) {
            if (arr[i] > suff[i]) {
                suff[i-1] = arr[i];
            } else {
                suff[i-1] = suff[i];
            }
        }
        return suff; 
    }
};

File: Data Structures & Algorithms/height-checker/submission-0.py
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        arr = [0]*101

        for h in heights:
            arr[h]+=1
        
        expected = []
        for i in range(len(arr)):
            for _ in range(arr[i]):
                expected.append(i)
        res = 0
        for i in range(len(expected)):
            if expected[i] != heights[i]:
                res+=1

        return res 


        